src/FixedRateIrm.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {IIrm} from "../lib/morpho-blue/src/interfaces/IIrm.sol";
+import {IFixedRateIrm} from "./interfaces/IFixedRateIrm.sol";
+
+import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import {Id, MarketParams, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+/* ERRORS */
+
+/// @dev Thrown when the rate is not already set for this market.
+string constant RATE_NOT_SET = "rate not set";
+/// @dev Thrown when the rate is already set for this market.
+string constant RATE_SET = "rate set";
+/// @dev Thrown when trying to set the rate at zero.
+string constant RATE_ZERO = "rate zero";
+
+/// @title FixedRateIrm
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+contract FixedRateIrm is IFixedRateIrm {
+    using MarketParamsLib for MarketParams;
+
+    /* EVENTS */
+
+    /// @notice Emitted when a borrow rate is set.
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    /* STORAGE */
+
+    /// @notice Borrow rates.
+    mapping(Id => uint256) public borrowRateStored;
+
+    /* SETTER */
+
+    /// @notice Sets the borrow rate for a market.
+    /// @dev A rate can be set by anybody, but only once.
+    /// @dev `borrowRate` reverts on rate not set, so the rate needs to be set before the market creation.
+    /// @dev The creator of a market with this IRM would typically batch setting of the rate with the market creation.
+    function setBorrowRate(Id id, uint256 newBorrowRate) external {
+        require(borrowRateStored[id] == 0, RATE_SET);
+        require(newBorrowRate != 0, RATE_ZERO);
+
+        borrowRateStored[id] = newBorrowRate;
+
+        emit SetBorrowRate(id, newBorrowRate);
+    }
+
+    /* BORROW RATES */
+
+    /// @inheritdoc IIrm
+    function borrowRateView(MarketParams memory marketParams, Market memory) external view returns (uint256) {
+        uint256 borrowRateCached = borrowRateStored[marketParams.id()];
+        require(borrowRateCached != 0, RATE_NOT_SET);
+        return borrowRateCached;
+    }
```

Comment:
I'm unsure if `borrowRateView` shouldn't just return `borrowRateStored`, without reverting. I don't think I see a clear use case given that `borrowRate` already has the reverting behavior

Reply:
I think that both should behave exactly the same way (like in the Adaptive curve irm)

Reply:
Ok fair, let's keep it like that